### PR TITLE
Add changes that support settings files

### PR DESCRIPTION
**Pull request by dedemorton**
Adds doc about settings file to resolve an item listed in  #5148 

Summary of changes:
- Formatting improvement requested by @acchen97 
- Add mention of the settings file as an option instead of specifying flags
- Fix a few minor mistakes and typos
- Add "coming" annotation to clarify that the changes are for alpha3
- Added node.name and pipeline.output.workers because it was missing from the doc (I'll add a note below about pipeline.output.workers)

### CODE REVIEW

File: docs/static/command-line-flags.asciidoc [hunk 1]
```diff
@@ -3,81 +3,91 @@
...
+ is specified, `stdout { codec => rubydebug }}` is default.
 
--w, --pipeline.workers COUNT
- Sets the number of pipeline workers (threads) to run for filter processing (default: number of cores).
+*`-w, --pipeline.workers COUNT`*::
+ Sets the number of pipeline workers (threads) to run for filter processing (default: half the number of cores).
  If you find that events are backing up, or that the CPU is not saturated, consider increasing
  this number to better utilize machine processing power.
+ 
+*`--pipeline.output.workers COUNT`*::
```
Comment (dedemorton) [inline docs/static/command-line-flags.asciidoc]: @suyograo or @jsvd I'm not convinced this is a valid CLI flag.

pipeline.output.workers DOES NOT exist here: 
https://github.com/elastic/logstash/blob/master/logstash-core/lib/logstash/runner.rb

But it does exist here:
https://github.com/elastic/logstash/blob/master/config/logstash.yml#L34

**Comment by suyograo** [comment]
@dedemorton Can we also add a separate section "Using the Settings File" or something similar to call out the existence of `config/logstash.yml` in detail? We can then reference this from the Command Line  section: https://github.com/elastic/logstash/pull/5332/files#diff-ab5ecd26344a9aea61a0a0581e8f283cR6

Since this is new its good to push its visibility.

File: docs/static/command-line-flags.asciidoc (hunk 1 above)
Comment (suyograo) [inline docs/static/command-line-flags.asciidoc]: @jsvd did you mean `workers` in `output` plugins?

Comment (suyograo) [inline docs/static/command-line-flags.asciidoc]: good catch @dedemorton

Comment (jsvd) [inline docs/static/command-line-flags.asciidoc]: @andrewvc should we expose `pipeline.output.workers` to the command line and make it known?

File: docs/static/command-line-flags.asciidoc [hunk 2]
```diff
@@ -3,81 +3,91 @@
...
  Use the given string as the configuration data. Same syntax as the config file.
- If not input is specified, 'stdin { type => stdin }' is default. If no output
- is specified, 'stdout { codec => rubydebug }}' is default.
+ If no input is specified, `stdin { type => stdin }` is default. If no output
+ is specified, `stdout { codec => rubydebug }}` is default.
 
--w, --pipeline.workers COUNT
- Sets the number of pipeline workers (threads) to run for filter processing (default: number of cores).
+*`-w, --pipeline.workers COUNT`*::
+ Sets the number of pipeline workers (threads) to run for filter processing (default: half the number of cores).
```
Comment (jsvd) [inline docs/static/command-line-flags.asciidoc]: I mislead you here, @dedemorton, it's actually the number of cores, not half (set in [here](https://github.com/elastic/logstash/blob/7783f76a4c29e2d944c807ceea1b4df85383c423/logstash-core/lib/logstash/environment.rb#L18))

File: docs/static/command-line-flags.asciidoc (hunk 1 above)
Comment (suyograo) [inline docs/static/command-line-flags.asciidoc]: IMHO, we should not. Its just inconsistent to have 1 config option being exposed this way. We should not set this precedent.

**Comment by dedemorton** [comment]
@suyograo Sure, I can add the info to a separate topic. I decided to err on the side of minimalism here, but I can see how having a separate topic might raise visibility that the file exists.

**Comment by jsvd** [comment]
LGTM

**Comment by suyograo** [comment]
LGTM

**Pull request closed without merging**